src/bicep-registry/scheduled-action/main.bicep

```diff
@@ -0,0 +1,138 @@
+// @resourceGroup @subscription @tenant
+
+// See https://learn.microsoft.com/rest/api/cost-management/scheduled-actions/create-or-update
+
+//===| Parameters |============================================================
+
+targetScope = 'subscription'
+
+@description('Name of the scheduled action used in the resource ID.')
+param name string
+
+@description('Indicates the kind of scheduled action.')
+@allowed([
+  'Email'
+  'InsightAlert' // @subscription
+])
+param kind string = 'Email'
+
+// @tenant
+//   @description('The Azure resource ID for the billing scope to use. This is the full resource ID of a billing account, billing profile, department, etc. Do not use the GUID only.')
+//   param scope string
+
+@description('Specifies which built-in view to use. This is a shortcut for the full view ID.')
+@allowed([
+  ''
+  'AccumulatedCosts'
+  'DailyCosts'
+  'InvoiceDetails'
+])
+param builtInView string = ''
+
+@description('Required if kind is "Email" and builtInView is not set. The resource ID of the view to which the scheduled action will send. The view must either be private (tenant level) or owned by the same scope as the scheduled action. Ignored if kind is "InsightAlert" or if builtInView is set.')
+param viewId string = ''
+
+@description('The display name to show in the portal when viewing the list of scheduled actions.')
+param displayName string = name
+
+@description('The status of the scheduled action.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param status string = 'Enabled'
+
+@description('Email address of the person or team responsible for this scheduled action. This email address will be included in emails.')
+param notificationEmail string = ''
+
+@description('List of email addresses that should receive emails. At least one valid email address is required.')
+param emailRecipients array
+
+@description('The subject of the email that will be sent to the email recipients.')
+@maxLength(70)
+param emailSubject string = ''
+
+@description('Include a message for recipients to add context about why they are getting the email, what to do, and/or who to contact.')
+@maxLength(250)
+param emailMessage string = ''
```

Comment:
I don't see a way to specify this in the existing Scheduled Actions API am I missing something or this is forward looking?
Consider though that people will want a richer message capability at some point. Might be worth reaching out to MEO for alignment / collaboration.

Reply:
API supports message, [https://learn.microsoft.com/en-us/rest/api/cost-management/scheduled-actions/create-or-update-by-scope?tabs=HTTP#notificationproperties](https://learn.microsoft.com/en-us/rest/api/cost-management/scheduled-actions/create-or-update-by-scope?tabs=HTTP#notificationproperties)